ramips: add support for Asus RT-AC1200

w43322 opened the pull request:
Asus RT-AC1200 is a 2.4/5GHz dual band AC router,
based on MediaTek MT7628AN.

Specification:
* SoC: MT7628AN
* RAM: DDR2 64 MiB
* Flash: 16 MiB NOR (W25Q128BV)
* Wi-Fi:
  * 2.4GHz: SoC Built-in
  * 5GHz: MT7612EN
* Ethernet: 5x 100Mbps
  * Switch: SoC built-in
* USB: 1x 2.0

Flash Layout:
 0x0000000-0x0030000 : "bootloader"
 0x0030000-0x0040000 : "nvram"
 0x0040000-0x0050000 : "factory"
 0x0050000-0x1000000 : "firmware"

MAC address:
 LAN: factory 0x28
 WAN: factory 0x22
 2.4G: factory 0x4
 5G: factory 0x8004

Installation via **recovery** mode:
1. Download the Asus recovery firmware (windows) tool from 
   http://dlcdnet.asus.com/pub/ASUS/LiveUpdate/Release/Wireless/Rescue.zip
2. Set your ethernet IP manually 192.168.1.5 / 255.255.255.0 with NO
   gateway.
3. Plug in your ethernet to LAN port 1 on the router.
4. Load up the recovery software with the firmware file, but don't press
   "Upload" yet.
5. Plug in the router to power WHILE HOLDING the reset button in. While
   CONTINUING to hold the button, select "Upload" Continue to hold the
   reset button in until it finishes and verifies!
6. If that doesn't work try pressing "Upload" first just before you do
   step 5. At some point while holding reset the rescue tool will finally
   detect and upload the firmware. That's when you can let go of the
   reset button.
7. The router will reboot and not much will happen. Wait a minute or 2.
8. Power off and on the router again. Voila. Set everything your Ethernet 
   IP back to DHCP (automatically) and you're good to go.

Revert to stock firmware:
1. Install stock image via recovery mode.

Tested-by: Ivan Pavlov <AuthorReflex@gmail.com>
Signed-off-by: Ray Wang <raywang777@foxmail.com>

therealsummoner commented:
Question regarding LEDs: are the LAN/WAN LEDs blinking in the same "rhythm" as the Wifi LEDs, or are they blinking like crazy, even when there is no traffic?

w43322 commented:
> Question regarding LEDs: are the LAN/WAN LEDs blinking in the same "rhythm" as the Wifi LEDs, or are they blinking like crazy, even when there is no traffic?

LAN/WAN only blinks when there's traffic (in rhythm), and Wifi LEDs are only on if Wireless is enabled (they don't indicate traffic).

therealsummoner commented:
Nice! Wanted to ask, because I'm working on the RT-AC1200-V2 and here when the switch LEDs are not included in the gpio state_default group, they will be controlled internally and constant crazy-fast blinking is on loose.

w43322 commented:
> Nice! Wanted to ask, because I'm working on the RT-AC1200-V2 and here when the switch LEDs are not included in the gpio state_default group, they will be controlled internally and constant crazy-fast blinking is on loose.

Well, don't really know why that would happen with the v2 board, but good luck!

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,173 @@
+		usbpower {
+			gpio-export,name = "usbpower";
+			gpio-export,output = <1>;
+			gpios = <&gpio 6 GPIO_ACTIVE_HIGH>;
```
Fixed in the latest commit, thanks for your help!

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,173 @@
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x28>;
```
Fixed in the latest commit, thanks for your help!

w43322 replied:
Hi, after this change, the LAN mac becomes a random string; I've checked other 7628an .dts files and they all define the mac-address using the original way. May I ask why was this change necessary?

w43322 replied:
Never mind, I found out why. I was using 21.0.1 locally.

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,173 @@
+};
+
+&wmac {
+	status = "okay";
```
Fixed in the latest commit, thanks for your help!

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,173 @@
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
```
Fixed in the latest commit, thanks for your help!

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,173 @@
+		groups = "gpio", "i2s", "refclk", "spi cs1", "uart1", "wled_an";
+		function = "gpio";
+	};
+};
```
Fixed in the latest commit, thanks for your help!

musashino205 commented on target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:
```diff
@@ -34,6 +34,10 @@ ramips_setup_interfaces()
 		;;
+	asus,rt-ac1200)
+		ucidef_add_switch "switch0" \
+			"1:lan" "2:lan" "3:lan" "4:lan" "0:wan" "6@eth0"
```
This configuration is the same with hilink,hlk-7628n, hilink,hlk-7688a, etc. Please merge with those models.

```
	hilink,hlk-7628n|\
	hilink,hlk-7688a|\
	hiwifi,hc5861b|\
	...
	wrtnode,wrtnode2p|\
	wrtnode,wrtnode2r|\
	zyxel,keenetic-extra-ii)
		ucidef_add_switch "switch0" \
			"1:lan" "2:lan" "3:lan" "4:lan" "0:wan" "6@eth0"
		;;
```

w43322 replied:
fixed

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,187 @@
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```
AFAIK the `spi-max-frequency` on mt7628 also has an effect on mtd speed. You can try 40~50MHz and test it with `time dd if=/dev/mtd3 of=/dev/null`.

w43322 replied:
tried 40MHz, was stable with that; 50MHz was not very stable.

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,187 @@
+			};
+
+			partition@30000 {
+				label = "u-config";
```
Does the stock FW really use the partition name "u-config"? I'd expect something like "u-boot-env" or "config"...

w43322 replied:
From stock firmware boot log:
`Aug  1 00:00:34 kernel: #add mtd partition#`
`Aug  1 00:00:34 kernel: Creating 7 MTD partitions on "raspi":`
`Aug  1 00:00:34 kernel: 0x000000000000-0x000000030000 : "Bootloader"`
`Aug  1 00:00:34 kernel: 0x000000030000-0x000000040000 : "nvram"`
`Aug  1 00:00:34 kernel: 0x000000040000-0x000000050000 : "Factory"`
`Aug  1 00:00:34 kernel: 0x000000050000-0x000000f00000 : "linux"`
`Aug  1 00:00:34 kernel: 0x00000017b840-0x000000f00000 : "rootfs"`
`Aug  1 00:00:35 kernel: 0x000000f00000-0x000001000000 : "jffs2"`
`Aug  1 00:00:35 kernel: 0x000000000000-0x000001000000 : "ALL"`

w43322 replied:
I will change partition names accordingly.

w43322 replied:
fixed

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,187 @@
+		led-upgrade = &led_power;
+	};
+
+	chosen {
```
fixed

w43322 commented:
accidentally closed.. sorry

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,183 @@
+		};
+		
+		all {
+			label = "blue:all";
```
Is it an actual LED or a gpio line which turns on/off all the other LEDs?

w43322 replied:
This GPIO is the master switch to turn on/off all LEDs except for the power LED

w43322 replied:
Thanks, I'll look into it.

w43322 replied:
```
	gpio_export {
		compatible = "gpio-export";
		#size-cells = <0>;

		led_all {
			gpio-export,name = "led_all";
			gpio-export,output = <0>;
			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
		};
	};
```
Ended up declaring it this way, because if not defined it will default to disabling all LEDs, and it would be ideal to have LEDs during boot.

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,183 @@
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+	};
```
You can now remove this node entirely.

w43322 replied:
Used for declaration of `led_all`

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,183 @@
+};
+
+&pcie0 {
+	mt76@0,0 {
```
This should be:

```diff
 &pcie0 {
-	mt76@0,0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
 		reg = <0x0000 0 0 0 0>;
```

w43322 replied:
fixed

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,183 @@
+
+&state_default {
+	gpio {
+		groups = "gpio", "i2s", "refclk", "spi cs1", "uart1", "wled_an";
```
If you want to make the ethernet and wifi leds programmable, then this should be `groups = "i2c", "spi cs1", "refclk", "p0led_an", "p1led_an", "p2led_an", "p3led_an", "p4led_an", "wled_an";`.
If you want to let the hardware control the leds automatically, then `groups = "i2c", "spi cs1", "refclk";` will suffice (and you can drop all the lan/wan/wlan led settings).

mans0n replied:
> However both 5g and 2g wifi LEDs doesn't work correctly; the 2g led doesn't light up at all, and the 5g led keeps blinking when there's no activity (even when it's not enabled).

Hmm, it should not affect the 5g led as far as the led node in &pcie0 is kept...

Could you also try this?

```diff
 &state_default {
 	gpio {
 		...
 	};
 
+	wled_an {
+		groups = "wled_an";
+		function = "wled_an";
+	};
 };
```

w43322 replied:
Turns out the 5G-Wifi LED was working as intended. I thought it was randomly blinking but turns out it's one of my devices trying to connect to the router.
And I defined the 2G-Wifi LED this way: 
```
		wlan2g {
			label = "blue:wlan2g";
			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
			linux,default-trigger = "phy0tpt";
		};
```
`groups = "i2c", "spi cs1", "refclk", "wled_an";`
, so `01_led` will be left unchanged.

mans0n commented on target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:
```diff
@@ -69,6 +69,7 @@ ramips_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"1:lan" "0:wan" "6@eth0"
 		;;
+	asus,rt-ac1200|\
```
Please move this block to follow the alphabetical order.

w43322 replied:
fixed

mans0n commented:
Please wrap your commit message up to 75 characters per line.

> Switch:
>  WAN: 0x1
>  LAN1: 0x2
>  LAN2: 0x4
>  LAN3: 0x8
>  LAN4: 0x10

Could you rephrase it in decimal?

w43322 committed:
ramips: add support for Asus RT-AC1200

Asus RT-AC1200 is a 2.4/5GHz dual band AC router,
based on MediaTek MT7628AN.

Specification:
* SoC: MT7628AN
* RAM: DDR2 64 MiB
* Flash: 16 MiB NOR (W25Q128BV)
* Wi-Fi:
  * 2.4GHz: SoC Built-in
  * 5GHz: MT7612EN
* Ethernet: 5x 100Mbps
  * Switch: SoC built-in
* USB: 1x 2.0

Flash Layout:
 0x0000000-0x0030000 : "bootloader"
 0x0030000-0x0040000 : "nvram"
 0x0040000-0x0050000 : "factory"
 0x0050000-0x1000000 : "firmware"

MAC address:
 LAN: factory 0x28
 WAN: factory 0x22
 2.4G: factory 0x4
 5G: factory 0x8004

Installation via **recovery** mode:
1. Download the Asus recovery firmware (windows) tool from
   http://dlcdnet.asus.com/pub/ASUS/LiveUpdate/Release/Wireless/Rescue.zip
2. Set your ethernet IP manually 192.168.1.5 / 255.255.255.0 with NO
   gateway.
3. Plug in your ethernet to LAN port 1 on the router.
4. Load up the recovery software with the firmware file, but don't press
   "Upload" yet.
5. Plug in the router to power WHILE HOLDING the reset button in. While
   CONTINUING to hold the button, select "Upload" Continue to hold the
   reset button in until it finishes and verifies!
6. If that doesn't work try pressing "Upload" first just before you do
   step 5. At some point while holding reset the rescue tool will finally
   detect and upload the firmware. That's when you can let go of the
   reset button.
7. The router will reboot and not much will happen. Wait a minute or 2.
8. Power off and on the router again. Voila. Set everything your Ethernet
   IP back to DHCP (automatically) and you're good to go.

Revert to stock firmware:
1. Install stock image via recovery mode.

Tested-by: Ivan Pavlov <AuthorReflex@gmail.com>
Signed-off-by: Ray Wang <raywang777@foxmail.com>

AuthorReflex commented:
with PR https://github.com/openwrt/openwrt/pull/5044 and adding node 
	mediatek,portmap = <4>;
lan leds work as expected.

You can add 
Tested-by: Ivan Pavlov <AuthorReflex@gmail.com>
to commit message

AuthorReflex commented:
> 
> Will do, but it‘s `mediatek,led_source` not `portmap` , right?
I add new binding.

```

&esw {
	mediatek,portmap = <0x3e>;

	mediatek,led_source = <4>;
};
```

mans0n commented:
Please squash your changes into a single commit.
You can remove the blank line between Tested-by and Signed-off-by.

w43322 commented:
> Please squash your changes into a single commit. You can remove the blank line between Tested-by and Signed-off-by.

OK, commits squashed and empty line removed. :)

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,160 @@
+		};
+	};
+
+	gpio_export {
```
The convention is dash for node names, underscore for label names.

AuthorReflex replied:
This syntax is same as other similar devices, e.g. Keenetic-extra-ii

mans0n replied:
Yes, but we can do it better here.

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,160 @@
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		led_all {
```
Ditto.

AuthorReflex replied:
Ditto

w43322 replied:
Hi, is there a naming convention for property `gpio-export,name` under this node? The naming scheme in the current device trees seems all over the place; some has underscores, some has dashes, ans some has none:
```
PS C:\OpenWRT\github\openwrt\target\linux\ramips\dts> findstr "gpio-export,name" *.*
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "modem-enable";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "modem-rf-enable";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "rtc-enable";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "sim-select";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "sim1-detect";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "sim2-detect";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "modem-enable";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "modem-rf-enable";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "sim-select";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "sim1-detect";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "sim2-detect";
mt7620a_dlink_dwr-118-a1.dts:                   gpio-export,name = "usb";
mt7620a_dlink_dwr-118-a2.dts:                   gpio-export,name = "usb";
mt7620a_dovado_tiny-ac.dts:                     gpio-export,name = "usbpower";
mt7620a_edimax_br-6478ac-v2.dts:                        gpio-export,name="usb-power";
mt7620a_hiwifi_hc5761.dts:                      gpio-export,name = "usbpower";
mt7620a_hiwifi_hc5861.dts:                      gpio-export,name = "usbpower";
mt7620a_hiwifi_hc5861.dts:                      gpio-export,name = "sdpower";
mt7620a_hnet_c108.dts:                  gpio-export,name = "power_modem";
mt7620a_lava_lr-25g001.dts:                     gpio-export,name = "usbpower";
mt7620a_lenovo_newifi-y1s.dts:                  gpio-export,name = "usb0";
mt7620a_lenovo_newifi-y1s.dts:                  gpio-export,name = "usb1";
mt7620a_lenovo_newifi-y1s.dts:                  gpio-export,name = "usb2";
mt7620a_sanlinking_d240.dts:                    gpio-export,name = "power_mpcie2";
mt7620a_sanlinking_d240.dts:                    gpio-export,name = "power_mpcie1";
mt7620a_sercomm_na930.dts:                      gpio-export,name = "telit";
mt7620a_sitecom_wlr-4100-v1-002.dts:                    gpio-export,name = "usb-power";
mt7620a_tplink_archer-mr200.dts:                        gpio-export,name = "power_usb1";
mt7620a_yukai_bocco.dts:                        gpio-export,name = "rec";
mt7620a_yukai_bocco.dts:                        gpio-export,name = "play";
mt7620a_zyxel_keenetic-viva.dts:                        gpio-export,name = "usb";
mt7620n_dlink_dwr-921-c1.dts:                   gpio-export,name = "lte_modem_enable";
mt7620n_dlink_dwr-922-e2.dts:                   gpio-export,name = "lte_modem_enable";
mt7620n_zyxel_keenetic-omni-ii.dts:                     gpio-export,name = "usb";
mt7620n_zyxel_keenetic-omni.dts:                        gpio-export,name = "usb";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "m2-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie0-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie0-reset";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie1-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie1-reset";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie2-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie2-reset";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "pcie-perst-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "sim-select";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "watchdog-enable";
mt7621_d-team_newifi-d2.dts:                    gpio-export,name = "power_usb3";
mt7621_d-team_pbr-m1.dts:                       gpio-export,name = "power_usb2";
mt7621_d-team_pbr-m1.dts:                       gpio-export,name = "power_usb3";
mt7621_d-team_pbr-m1.dts:                       gpio-export,name = "power_sata";
mt7621_humax_e10.dts:                   gpio-export,name = "power:usb";
mt7621_lenovo_newifi-d1.dts:                    gpio-export,name = "usb2power";
mt7621_lenovo_newifi-d1.dts:                    gpio-export,name = "usb3power";
mt7621_mikrotik_routerboard-7xx.dtsi:                   gpio-export,name = "buzzer";
mt7621_mikrotik_routerboard-7xx.dtsi:                   gpio-export,name = "usb_power";
mt7621_sercomm_na502.dts:                       gpio-export,name = "zwave_reset";
mt7621_sercomm_na502.dts:                       gpio-export,name = "zigbee_reset";
mt7621_tplink_eap235-wall-v1.dts:                       gpio-export,name = "poe-passthrough";
mt7621_unielec_u7621-01.dtsi:                   gpio-export,name = "modem_reset";
mt7621_unielec_u7621-06.dtsi:                   gpio-export,name = "modem_reset";
mt7621_xzwifi_creativebox-v1.dts:                       gpio-export,name = "power_usb2";
mt7621_xzwifi_creativebox-v1.dts:                       gpio-export,name = "power_usb3";
mt7621_xzwifi_creativebox-v1.dts:                       gpio-export,name = "power_sata";
mt7621_zbtlink_zbt-wg1602.dtsi:                 gpio-export,name = "4g1-pwr";
mt7621_zbtlink_zbt-wg1602.dtsi:                 gpio-export,name = "4g2-pwr";
mt7621_zbtlink_zbt-wg1602.dtsi:                 gpio-export,name = "ext-usb";
mt7628an_asus_rt-ac1200.dts:                    gpio-export,name = "led_all";
mt7628an_glinet_gl-mt300n-v2.dts:                       gpio-export,name = "usb";
mt7628an_hak5_wifi-pineapple-mk7.dts:                   gpio-export,name = "usb-power";
mt7628an_hiwifi_hc5761a.dts:                    gpio-export,name = "usb_power";
mt7628an_minew_g1-c.dts:                        gpio-export,name = "ws2812";
mt7628an_minew_g1-c.dts:                        gpio-export,name = "nrf_power";
mt7628an_zyxel_keenetic-extra-ii.dts:                   gpio-export,name = "usbpower";
rt3050_arcwireless_freestation5.dts:                    gpio-export,name = "poe-passthrough";
rt3052_accton_wr6202.dts:                       gpio-export,name = "usb";
rt3052_jcg_jhr-n926r.dts:                       gpio-export,name = "display_data";
rt3052_jcg_jhr-n926r.dts:                       gpio-export,name = "display_clock";
rt3052_jcg_jhr-n926r.dts:                       gpio-export,name = "display_blank";
rt3052_planex_mzk-wdpr.dts:                     gpio-export,name = "lcd_ctrl1";
rt3662_dlink_dir-645.dts:                       gpio-export,name = "usb";
rt3662_edimax_br-6475nd.dts:                    gpio-export,name="usb";
rt3662_engenius_esr600h.dts:                    gpio-export,name = "usb";
rt3662_omnima_hpm.dts:                  gpio-export,name = "usb0";
rt3662_omnima_hpm.dts:                  gpio-export,name = "usb1";
rt3883_sitecom_wlr-6000.dts:                    gpio-export,name = "usb";
rt5350_dlink_dir-320-b1.dts:                    gpio-export,name = "usb";
rt5350_dlink_dir-320-b1.dts:                    gpio-export,name = "root_hub";
rt5350_dlink_dwr-512-b.dts:                     gpio-export,name = "slic_int";
rt5350_dlink_dwr-512-b.dts:                     gpio-export,name = "modem3g_enable";
rt5350_hame_mpr-a1.dts:                 gpio-export,name = "usb";
rt5350_hame_mpr-a1.dts:                 gpio-export,name = "root_hub";
rt5350_hame_mpr-a2.dts:                 gpio-export,name = "usb";
rt5350_hame_mpr-a2.dts:                 gpio-export,name = "root_hub";
rt5350_hilink_hlk-rm04.dts:                     gpio-export,name = "hlk-rm04:gpio0";
rt5350_hilink_hlk-rm04.dts:                     gpio-export,name = "hlk-rm04:gpio1";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio0";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio1";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio17";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio18";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio19";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio20";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio21";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio22";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio23";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio24";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio25";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio26";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio27";
rt5350_olimex_rt5350f-olinuxino-evb.dts:                        gpio-export,name = "relay1";
rt5350_olimex_rt5350f-olinuxino-evb.dts:                        gpio-export,name = "relay2";
rt5350_olimex_rt5350f-olinuxino-evb.dts:                        gpio-export,name = "button";
rt5350_poray_x5.dts:                    gpio-export,name = "usb-mode";
rt5350_tenda_3g150b.dts:                        gpio-export,name = "usb";
rt5350_trendnet_tew-714tru.dts:                 gpio-export,name = "repeater_switch";
rt5350_trendnet_tew-714tru.dts:                 gpio-export,name = "wisp_switch";
rt5350_unbranded_a5-v11.dts:                    gpio-export,name = "usb";
rt5350_unbranded_a5-v11.dts:                    gpio-export,name = "root_hub";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio0";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio7";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio8";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio9";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio12";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio13";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio14";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio17";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio18";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio19";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio20";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio21";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio22";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio23";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio24";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio25";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio26";
```

mans0n replied:
AFAIK there's no such thing. The convention I said above is from [1], but it is not that everyone is strictly following the convention (even the upstream kernel is mixed up), and it does not say anything on property value strings anyway. No one did not proposed and enforced a good convention (yet), that's all.
Personally, I prefer dash to underscore in all cases except where dash cannot be used (e.g. label). [2]

[1] https://elinux.org/Device_Tree_Linux#Linux_conventions
[2] https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3

w43322 replied:
Got it, thanks. I'll change it to a dash for the sake of consistency in the next (and hopefully final) commit, and (hopefully) address the LED blinking issue along with it.

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,160 @@
+
+&esw {
+	mediatek,portmap = <0x3e>;
+	mediatek,led_source = <4>;
```
This line makes the PR blocked by #5044...
BTW I've just opened a new PR #9282, I would be glad if you can test it on your device.

AuthorReflex replied:
PR #9282 tested. It's working better than PR #5044

AuthorReflex replied:
for PR#9282 need to set "mediatek,eee-disable" instead

w43322 replied:
Yes, I will add once PR#9282 is merged.

mans0n commented on target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts:
```diff
@@ -0,0 +1,160 @@
+	};
+};
+
+&factory {
+	compatible = "nvmem-cells";
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	macaddr_factory_28: macaddr@28 {
+		reg = <0x28 0x6>;
+	};
+};
```
This can be moved into the factory partition node.

AuthorReflex replied:
Same as similar devices

mans0n replied:
example of consolidation: 256e1dbc7d09

mans0n reviewed (commented):
I hope this will be the last round of the review.

w43322 reviewed (commented):
To avoid merge block, I've deleted the line regarding the switch LED binding. I will add it back once the issue is resolved in the code base.

mans0n commented:
Thanks! Merged your patch as 3c4810fecc.
